discussions/prog_repo.md

```diff
@@ -0,0 +1,197 @@
+# L3AF Kernel Function Marketplace
+
+The concept of a L3AF Kernel Function Marketplace is to create a location where
+Kernel Functions from any trusted party can be uploaded and made available for
+others to download.
+
+In the context of L3AF, we define a Kernel Function a kernel eBPF program with
```

Comment:
The name "Kernel Function" for this concept is probably a misnomer. Please discuss alternative names here.

Reply:
We'd like this name to (a) not include "kernel", (b) not be limited to "functions", (c) be quick, easy, descriptive, (d) not reinvent the wheel for the sake of being "cute" or "clever". Some ideas off the top of my head…

* App: commonly used term; does it imply a self-contained thing or would it work as well for chainables?
* Package: works with `package manager`; implies it's a thing that contains other things, which these will be? I think?

However, this brings up a question: When you look at one of these things that the package manager, uh, manages, what do you see? Is it a collection of related files or a definition/manifest that can build the collection or…? Perhaps we should get a shared/common idea of _what's_ being distributed before we go much further on _how_ they get distributed?

Reply:
`eBPF Package` seems like a reasonable name imo

Reply:
"ebpf package" is fine with me for the thing that can include both app code and ebpf program code.

Reply:
I agree,  we can probably look to replace "Kernel Function" with "eBPF Program" and we can also call this "repository".. which would make this eBPF Program Repository.

Reply:
Catching up on the above comments - - We can call it `eBPF package`; the reason I preferred the name `eBPF program` is that this repo may contain many simple eBPF programs that leverage kprobes, uprobes, and tracepoints. And so, an eBPF program will really be the LCD here.